### ISSUE TITLE
[Testing] Test that dask chunks are all the same shape

### ISSUE THREAD

**Issue by jhamman** [jhamman/xpublish#15]
Zarr uses uniform chunk sizes. As long as we're using dask's `.blocks` to access data, we need an explicit check to make sure all blocks (except edge chunks) are equal shapes.

**Comment by andersy005** [comment]
I meant to ask you about this last night. So, if let's you have

```python
In [1]: import dask

In [2]: import dask.array as da

In [3]: x  = da.arange(5, chunks=(2,))

In [4]: x
Out[4]: dask.array<arange, shape=(5,), dtype=int64, chunksize=(2,), chunktype=numpy.ndarray>
```

It's my understanding that zarr will write the last chunk to be of shape (2,) as well, and when reading the chunk back, zarr will discard the undesired bytes. How does zarr accomplish this?

**Comment by jhamman** [comment]
Closed via #16.
